app/models/service_retire_task.rb

```diff
@@ -14,5 +14,47 @@ def update_and_notify_parent(*args)
   end
 
   def task_finished
+    update_attributes(:status => status == 'Ok' ? 'Completed' : 'Failed')
+  end
+
+  def task_active
+    update_attributes(:status => 'Active')
+  end
+
+  def after_request_task_create
+    update_attributes(:description => get_description)
+    parent_svc = Service.find_by(:id => options[:src_ids])
+    _log.info("- creating service tasks for service <#{self.class.name}:#{id}>")
+
+    create_retire_subtasks(parent_svc)
+  end
+
+  def create_retire_subtasks(parent_service)
+    parent_service.service_resources.collect do |svc_rsc|
+      next unless svc_rsc.resource.present? && svc_rsc.resource.respond_to?(:retire_now)
+      next if svc_rsc.resource_type == "ServiceTemplate"
+
+      nh = attributes.except("id", "created_on", "updated_on", "type", "state", "status", "message")
+      nh['options'] = options.except(:child_tasks)
+      # Initial Options[:dialog] to an empty hash so we do not pass down dialog values to child services tasks
+      nh['options'][:dialog] = {}
+      next if svc_rsc.resource_type == "ServiceTemplate" &&
```

Comment:
Can this be moved up closer to the other `next`s?  If so, it could potentially avoid allocating extra objects for `nh`